src/xcode/ENA/ENA/Source/SceneDelegate.swift

```diff
@@ -296,14 +304,20 @@ extension SceneDelegate: UNUserNotificationCenterDelegate {
 	}
 
 	func userNotificationCenter(_: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
-		switch response.actionIdentifier {
-		case UserNotificationAction.openExposureDetectionResults.rawValue,
-			 UserNotificationAction.openTestResults.rawValue:
+		switch response.notification.request.identifier {
+		case ActionableNotificationIdentifier.testResult.identifier,
+			 ActionableNotificationIdentifier.riskDetection.identifier,
+			 ActionableNotificationIdentifier.deviceTimeCheck.identifier:
 			showHome(animated: true)
-		case UserNotificationAction.ignore.rawValue,
-			 UNNotificationDefaultActionIdentifier,
+			
+		case ActionableNotificationIdentifier.warnOthersReminder1.identifier,
+			 ActionableNotificationIdentifier.warnOthersReminder2.identifier:
+			 showPositiveTestResultFromNotification(animated: true)
+		
+		case UNNotificationDefaultActionIdentifier,
 			 UNNotificationDismissActionIdentifier:
 			break
+
```

Comment:
Why put everything in an extension? Isn't UNUserNotificationCenterDelegate some kind of protocol?
I think we should add default MARKs here and add move these extensions to designated section inside the MARKs

Reply:
Open for discussion 👍 . We took what was already there and followed the structure.

Reply:
Whoever touches it owns it 😄

Reply:
Restructured it now in a meeting with Karsten and Puneet